### PR TITLE
[WFLY-11829]: Support messaging clusters (Artemis Clusters) behind http load balancers.

### PR DESCRIPTION
**Pull request by ehsavoie**
* Adding a new attribute in 'all' cf and pcf to avoid sending the internal cluster topology to the clients.

Jira: https://issues.jboss.org/browse/WFLY-11829

### CODE REVIEW

**Comment by ehsavoie** [comment]
@jmesnil could you please take a look ?

File: messaging/EAP7-669_artemis_clusters_with_http_load_balancer.adoc [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/EAP7-669
```
Comment (bstansberry) [inline messaging/EAP7-669_artemis_clusters_with_http_load_balancer.adoc]: Please create a WFLY and make that the primary issue here. Please update the PR title and commit message accordingly.

File: messaging/EAP7-669_artemis_clusters_with_http_load_balancer.adoc [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+=== Other Interested Projects
+
+== Requirements
+
+* Add a boolean `use-topology-load-balancing` attribute to 
+- the https://wildscribe.github.io/WildFly/16.0/subsystem/messaging-activemq/connection-factory/[/subsystem=messaging-activemq/connection-factory resource]
+- the https://wildscribe.github.io/WildFly/16.0/subsystem/messaging-activemq/server/connection-factory[/subsystem=messaging-activemq/server=*/connection-factory resource]
+- the https://wildscribe.github.io/WildFly/16.0/subsystem/messaging-activemq/pooled-connection-factory/[/subsystem=messaging-activemq/pooled-connection-factory resource]
+- the https://wildscribe.github.io/WildFly/16.0/subsystem/messaging-activemq/server/pooled-connection-factory/[/subsystem=messaging-activemq/server=*/pooled-connection-factory resource]
+
```
Comment (bstansberry) [inline messaging/EAP7-669_artemis_clusters_with_http_load_balancer.adoc]: What are the default values of these attributes?

File: messaging/EAP7-669_artemis_clusters_with_http_load_balancer.adoc [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+=== Hard Requirements
+
+N/A
```
Comment (bstansberry) [inline messaging/EAP7-669_artemis_clusters_with_http_load_balancer.adoc]: What are the required effects of these settings?

File: messaging/EAP7-669_artemis_clusters_with_http_load_balancer.adoc (hunk 1 above)
Comment (rhusar) [reply messaging/EAP7-669_artemis_clusters_with_http_load_balancer.adoc]: +1 this is public repository but EAP7 issues are not accessible publicly.

**Comment by rhusar** [comment]
Would you mind filling in some actual information about the issue? Its not clear what `use-topology-load-balancing` does.

**Comment by bstansberry** [comment]
@ehsavoie No references to EAP7 JIRA except as links in the Related Issues section of the analysis.

File: messaging/EAP7-669_artemis_clusters_with_http_load_balancer.adoc (hunk 2 above)
Comment (ehsavoie) [reply messaging/EAP7-669_artemis_clusters_with_http_load_balancer.adoc]: fixed

File: messaging/EAP7-669_artemis_clusters_with_http_load_balancer.adoc (hunk 1 above)
Comment (ehsavoie) [reply messaging/EAP7-669_artemis_clusters_with_http_load_balancer.adoc]: https://issues.jboss.org/browse/WFLY-11829

**Comment by ehsavoie** [comment]
All is fixed

File: messaging/WFLY-11829_artemis_clusters_with_http_load_balancer.adoc [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+[ ] QE
+
+=== Affected Projects or Components
+
+=== Other Interested Projects
+
+== Requirements
+
+* Add a boolean `use-topology-load-balancing` attribute, with a default value set to `true`, to
```
Comment (pkremens) [inline messaging/WFLY-11829_artemis_clusters_with_http_load_balancer.adoc]: https://github.com/ehsavoie/wildfly/commit/966abbf56d0f5483f0360f11757ea00a69349293 this should be `use-topology-for-load-balancing`, could you please update here?

**Comment by fmarchioni** [comment]
The document is ok from my side.

**Comment by fmarchioni** [comment]
Thanks!

**Pull request merged**